 ## I.  Variables, names, assignment, updating variables, reassignment

###I. Practice creating variables here. For example,

`quiz01 = 90`

`quiz02 = 100`

`quizaverage = (quiz01 + quiz02)/2`


In [ ]:
#code here

quiz01 = 90

quiz02 = 100

quizaverage = (quiz01 + quiz02)/2

In [ ]:
print(quizaverage)


95.0


In [ ]:
quiz03

NameError: name 'quiz03' is not defined

In [ ]:
assignment = 30


In [ ]:
assignment01@nyu = 95

SyntaxError: cannot assign to expression here. Maybe you meant '==' instead of '='? (<ipython-input-11-72595c1e7a41>, line 1)

In [ ]:
class = 90.5 # reserved word

SyntaxError: invalid syntax (<ipython-input-12-f707e5b21683>, line 1)

In [ ]:
print(quizaverage)
print(quiz01)

95.0
90


In [ ]:
quiz01 == 90

True

In [ ]:
del quizaverage

In [ ]:
message = "Welcome to class 2." #Assign the value to a variable


In [ ]:
age = 27 # assignment operator

In [ ]:
age == 23 # comparision operator for equality

False

In [ ]:
lastname = "Smith"
lastname = lastname.upper() #function to convert strings to uppercase lettering
print(lastname) #prints out the result

SMITH


In [ ]:
lastname = "Sosulski"
name = lastname
#print(name)

In [ ]:
name = "Goss"

In [ ]:
lastname = "Sosulski-Goss"
print(lastname)
print(name)

Sosulski-Goss
Sosulski


In [ ]:
type(False)

bool

In [ ]:
customer_id = 12345
isinstance(customer_id, int)

True

In [ ]:
isinstance(lastname, str)

True

In [ ]:
myname = ""

In [ ]:
type(myname)

str

In [ ]:
isinstance(name,str)

True

In [ ]:
salary = 240,000 # this will cause a logical error.
salary

(240, 0)

In [ ]:
salary[0]

240

In [ ]:
salary = 240000
print("${:,.0f}".format(salary))

$240,000


In [ ]:
salary = 240000
print(f"${salary:,.0f}")

$240,000


In [ ]:
salary

240000

In [ ]:
lastname = "Sosulski"
lastname = lastname.upper()
lastname

'SOSULSKI'

In [ ]:
lastname.islower()

False

In [ ]:
first= "Julia"
last = "Jones"
space=" "
first+space+last

'Julia Jones'

In [ ]:
(first+space+last+space) * 4

'Julia Jones Julia Jones Julia Jones Julia Jones '

### Conditional example

In [ ]:
married = True

In [ ]:
if (married):
  print ("This customer is married.")
else:
  print ("This customer is not married.")


This customer is married.


In [ ]:
#throws an indentation error
if(True):
print("This is true")
else:
print("It is false")

IndentationError: expected an indented block after 'if' statement on line 2 (<ipython-input-70-67f75901e9b5>, line 3)

In [ ]:
if(3+5 == 7):
  print("It's true. It really is!")
else:
  print("It's not true")

It's not true


In [ ]:
mycondition = True

if(mycondition):
  print("It's True")
else:
  print("Must be False!")

It's True


In [ ]:

if (3+3 == 7):
  print("It's true")
else:
  print("It's false")

It's false


###II. Create the variables below.

Variable | Value
----------|------------
customer id |  12345
first name | Wanda
last name  | Wonderly
gender | Female
married | False
annual income | 240,000

In [ ]:
customer_id = 12345
first_name ="Wanda"
last_name = "Wonderly"
gender = "female"
married = False
annual_income = 240000

### III. Naming Variables.  Run the code for examples 1-3. Use the `#` to add comments to explain the output.

**Example 1.**

In [ ]:
76trombones = 'big parade'


SyntaxError: ignored

**Example 2.**

In [ ]:
more@ = 1000000


**Example 3.**

In [ ]:
class = 'Advanced Theoretical Zymurgy'

### IV. Number formatting

In [ ]:
x = 42
print(type(x))
print("This costs ${:.2f}".format(x))

<class 'int'>
This costs $42.00


In [ ]:
#alternative with fstrings

x = 42
print(type(x))
print(f"This costs ${x:.2f}")

<class 'int'>
This costs $42.00


### VI. Characters and ASCII

In [ ]:
# Program to find the ASCII value of a character

ch = "D"

print("The ASCII decimal value of char " + ch + " is:",ord(ch),sep=" ", end="\n")
print("Hi")


The ASCII decimal value of char D is: 68
Hi


In [ ]:
help(print)

Help on built-in function print in module builtins:

print(...)
    print(value, ..., sep=' ', end='\n', file=sys.stdout, flush=False)
    
    Prints the values to a stream, or to sys.stdout by default.
    Optional keyword arguments:
    file:  a file-like object (stream); defaults to the current sys.stdout.
    sep:   string inserted between values, default a space.
    end:   string appended after the last value, default a newline.
    flush: whether to forcibly flush the stream.



In [ ]:
chr(9748) # raining emoji in decimal

'☔'

In [ ]:
print("\U0001F600") #U+1F600 added 3 zeros after U

😀


In [ ]:
print("\U0001F929")

🤩


In [ ]:
print("\U0001F49d") # Happy Valentine's Day!

💝


In [ ]:
print("\U0001F3BE")

🎾


In [ ]:
print("\U0001F63F")

😿


In [ ]:
# Program to find the character value of a ASCII decimal.
num = 116
print("The character value of the ASCII decimal", num, "is:", chr(num), end=".")

The character value of the ASCII decimal 116 is: t.

In [ ]:
#Write your name in binary
myname ="Kristen"
print(f"The value of K in decimal is {ord(myname[0])}")
print(f"The binary representation of the letter {myname[0]} is {format(ord(myname[0]), '08b')}")

The value of K in decimal is 75
The binary representation of the letter K is 01001011


In [ ]:
ord(myname[0])

75

Let's work it out using our chart:


0 128 x 0 = 0

1  64 x 1 = 64

0  32 X 0 = 0

0  16 X 0 = 0

1  8  X 1 = 8

0  4  X 0 = 0

1  2  X 1 = 2

1  1  X 1 = 1

            64 + 8 + 2 +1 = 75

            BINARY: 01001011



In [ ]:
#Advanced: Using a for loop to print myname in binary
myname = "Kris"

for index in range(len(myname)):
  print(format(ord(myname[index]),'08b'), end=" ")

01001011 01110010 01101001 01110011 

In [ ]:
value = 100
format(value, '08b')

'01100100'

In [ ]:
chr(49)
ord("1")
format(ord("1"), '08b')

'00110001'

VII. Variable reassignment and changing data types

In [ ]:
hours = "5"

In [ ]:
type(hours)

str

In [ ]:
hours = int(hours)

In [ ]:
type(hours)

int

In [ ]:
hours = str(hours)

In [ ]:
type(hours)

str

In [ ]:
myname ="Kris"

In [ ]:
int(myname)

ValueError: ignored

In [ ]:
isinstance(myname,str)

True

### IX. Reading user input

####Example 1. Collect one value from a user

In [ ]:

input("Enter your name and press the enter key.")

In [ ]:
username = input("Enter your name and press the enter key.\n")
workout_hours = input("How many hours on average do you exercise per week.\n")
print(f"Hi username! You have spent {float(workout_hours)*52} working out this year.")

In [ ]:
type(workout_hours)

str

#####Example 2. Collect multiple values from a user and display output that computes the number of they work out per year.  

In [ ]:
username = input("Enter your name and press the enter key.\n")
workout_hours = input("Enter the number of hours you work per week.\n")
print(f"Congrats {username}! You have worked out {workout_hours*52} hours this year!")


Enter your name and press the enter key.
Kristen
Enter the number of hours you work per week.
2
Congrats Kristen! You have worked out 2222222222222222222222222222222222222222222222222222 hours this year!


In [ ]:
#Let's look at the var

print(workout_hours)
type(workout_hours)


10


str

In [ ]:
#What could we do to fix this?
username = input("Enter your name and press the enter key.\n")
workout_hours = input("How many hours on average do you exercise per week.\n")
#workout_hours * 52
workout_hours = float(workout_hours)
workout_hours * 52


Enter your name and press the enter key.
Kristen
How many hours on average do you exercise per week.
four


ValueError: ignored

In [ ]:

username = input("Enter your name and press the enter key.\n")
workout_hours = input("How many hours on average do you exercise per week.\n")
print("Hi", username,". You have spent", round(float(workout_hours)*52, 2), "working out this year.", sep=" ")

Enter your name and press the enter key.
k
How many hours on average do you exercise per week.
2
Hi k . You have spent 104.0 working out this year.


### X. The challenge:

Create a program that asks a user for their netid and number of hours they spent on lesson 1 in this course. Then print out a statement that multiples that number of remaining lessons (11) by the number hours spent on lesson 1.

#### Approach 1

In [ ]:
id =input("What is your netid? \n") # adding new line
hours = input("how many hours did you spend on lesson 1?\n") # adding new line
print("There are 11 more lessons. You should expect to spend at least", 11
      * float(hours), "hours between now and the last class.")

What is your netid? 
vs2896


KeyboardInterrupt: Interrupted by user

#### Approach 2

In [ ]:
#adding exception

id =input("What is your netid? \n") # adding new line
hours = input("how many hours did you spend on lesson 1?\n") # adding new line

try:
  hours=float(hours)
  print("There are 11 more lessons. You should expect to spend at least", 11 * hours, "hours between now and the last class.")
except:
  print("Please start over and enter a number.")

What is your netid? 
ks123
how many hours did you spend on lesson 1?
four
Please start over and enter a number.


#### Approach 3

In [ ]:

#using while loop

invalid = True
id =input("What is your netid? \n") # adding new line

while invalid:

  hours = input("how many hours did you spend on lesson 1?\n") # adding new line

  try:

    hours=float(hours)
    print("There are 11 more lessons. You should expect to spend at least", 11 * hours, "hours between now and the last class.")
    invalid = False
    #also could use a break here
  except:
    print("Your input:", hours, "is invalid. Please start over and enter a number.")

What is your netid? 
vs2896
how many hours did you spend on lesson 1?
fr
Your input: fr is invalid. Please start over and enter a number.
how many hours did you spend on lesson 1?
5
There are 11 more lessons. You should expect to spend at least 55.0 hours between now and the last class.


In [ ]:
#olivia's example
invalid = True
username = input("Enter your name and press the enter key.")

while invalid:

  workout_hours = input("Enter the number of hours you work per week.\n")

  try:
    workout_hours=float(workout_hours)
    print(f"Congrats {username}! You have worked out {round((workout_hours *52),2)} hours this year!")
    invalid = False
    #also could use a break here
  except:
    print("Your input:", hours, "is invalid. Please start over and enter a number.")

Enter your name and press the enter key.kristen
Enter the number of hours you work per week.
4
Congrats kristen! You have worked out 208.0 hours this year!


#### Approach 4 Interactive program demonstration

In [ ]:
invalid = True
id =input("What is your netid? \n") # adding new line
while invalid:

  hours = input("how many hours did you spend on lesson 1?\n") # adding new line

  try:
    hours=float(hours)
    print("There are 11 more lessons. You should expect to spend at least", 11 * hours, "hours between now and the last class.")
    invalid = False
    #also could use a break here
  except:
    print("Your input:", hours, "is invalid. Please start over and enter a number.")

What is your netid? 
v45
how many hours did you spend on lesson 1?
...
Your input: ... is invalid. Please start over and enter a number.
how many hours did you spend on lesson 1?
5.6
There are 11 more lessons. You should expect to spend at least 61.599999999999994 hours between now and the last class.


In [ ]:
mylist = [1,2,34,32]
mytuple = tuple(mylist)

## Write your name in binary.

In [ ]:
myname ="Kristen"
ord(myname[0])
format(ord(myname[0]), '08b') #displays just K

'01000001'

In [ ]:
# initializing string
test_str = "Kristen"

# printing original string
print("The original string is : " + str(test_str))

# using join() + ord() + format()
# Converting String to binary
res = ' '.join(format(ord(i), '08b') for i in test_str) #advanced topic on for loops. See lesson 5

# printing result
print("The string after binary conversion : " + str(res))

The original string is : Kristen
The string after binary conversion : 01001011 01110010 01101001 01110011 01110100 01100101 01101110
